mock/src/mock/chain.rs

```diff
@@ -6,7 +6,7 @@ use miden_objects::{
     crypto::merkle::{Mmr, NodeIndex, PartialMmr, SimpleSmt, TieredSmt},
     notes::{Note, NoteInclusionProof, NOTE_LEAF_DEPTH, NOTE_TREE_DEPTH},
     transaction::{ChainMmr, InputNote},
-    utils::collections::{BTreeMap, Vec},
+    utils::collections::Vec,
```

Comment:
`Vec` in prelude comes from `std`. We use our own import so that it automatically pulls it either from `std` or `alloc` depending on the context. This way, we can use the same import for both `std` and `no_std` context.